subprojects/core/src/main/java/org/gradle/api/internal/catalog/parser/TomlCatalogFileParser.java

```diff
@@ -79,31 +81,33 @@ public class TomlCatalogFileParser {
         "rejectAll"
     );
 
-    public static void parse(InputStream in, VersionCatalogBuilder builder) throws IOException {
+    public static void parse(Path catalogFilePath, VersionCatalogBuilder builder) throws IOException {
         StrictVersionParser strictVersionParser = new StrictVersionParser(Interners.newStrongInterner());
-        TomlParseResult result = Toml.parse(in);
-        assertNoParseErrors(result, builder);
-        TomlTable metadataTable = result.getTable(METADATA_KEY);
-        verifyMetadata(builder, metadataTable);
-        TomlTable librariesTable = result.getTable(LIBRARIES_KEY);
-        TomlTable bundlesTable = result.getTable(BUNDLES_KEY);
-        TomlTable versionsTable = result.getTable(VERSIONS_KEY);
-        TomlTable pluginsTable = result.getTable(PLUGINS_KEY);
-        Sets.SetView<String> unknownTle = Sets.difference(result.keySet(), TOP_LEVEL_ELEMENTS);
-        if (!unknownTle.isEmpty()) {
-            throwVersionCatalogProblem(builder, VersionCatalogProblemId.TOML_SYNTAX_ERROR, spec ->
-                spec.withShortDescription(() -> "Unknown top level elements " + unknownTle)
-                    .happensBecause(() -> "TOML file contains an unexpected top-level element")
-                    .addSolution(() -> "Make sure the top-level elements of your TOML file is one of " + oxfordListOf(TOP_LEVEL_ELEMENTS, "or"))
-                    .documented());
+        try (InputStream inputStream = Files.newInputStream(catalogFilePath)) {
```

Comment:
:question: shouldn't this use a buffered stream? I don't know if `Toml` itself does any buffering.

Reply:
Toml does buffer the stream!
Marking this question as resolved.